### PR TITLE
Add support for structured decision models

### PR DESCRIPTION
**Pull request by vincevannoort**
# What does this PR do?

This pull request adds support for structured decision models. The [Laya model](https://github.com/NandhaKishorM/laya) is an open-source implementation by [the author](https://github.com/NandhaKishorM) of two papers linked in the references. The model is similar to the recent [JEV model](https://typesafe.ai/blog/introducing-system-one-models-and-jev) by TypeSafe AI.

Disclaimer: I did use AI to help me write this pull request.

<details>
<summary>Example request and output</summary>

Request
```curl
curl -X POST http://localhost:8080/v1/decide \
  -H 'Content-Type: application/json' \
  -d '{                                              
    "state": {
      "from": "user@acme.com",
      "subject": "Duplicate charge on invoice #4411",
      "body": "Hi, we were billed twice for March. Please refund the duplicate today or we will cancel our plan."
    },
    "questions": {
      "department": {
        "type": "choice",
        "instructions": "Which department should handle this request?",
        "criteria": {
          "billing": "invoices, payments, refunds",
          "technical": "bugs, outages, system errors",
          "sales": "pricing, new contracts",
          "other": "everything else"
        }
      },
      "urgency": {
        "type": "score",
        "instructions": "How urgent is this request?",
        "criteria": [
          "not urgent",
          "soon",
          "critical deadline or blocking issue"
        ]
      },
      "churn_risk": {
        "type": "noul",
        "instructions": "Does the user threaten to cancel or leave?"
      },
      "refund_requested": {
        "type": "noul",
        "instructions": "Does the user explicitly request a refund?"
      }
    }
  }'
```

Output
```json
{
    "answers": {
        "urgency": {
            "type": "score",
            "score": 1.1975098,
            "legend": {
                "0": "not urgent",
                "1": "soon",
                "2": "critical deadline or blocking issue"
            },
            "probabilities": {
                "0": 0.1977539,
                "1": 0.4074707,
                "2": 0.39501953
            },
            "confidence": 0.04130894,
            "action": 0,
            "action_probability": 1.0
        },
        "refund_requested": {
            "type": "noul",
            "noul": 0.8432617,
            "confidence": 0.8433838,
            "action": 0,
            "action_probability": 1.0
        },
        "churn_risk": {
            "type": "noul",
            "noul": 0.79785156,
            "confidence": 0.79785156,
            "action": 0,
            "action_probability": 1.0
        },
        "department": {
            "type": "choice",
            "label": "billing",
            "probabilities": {
                "billing": 0.94140625,
                "other": 0.014404297,
                "sales": 0.02558899,
                "technical": 0.018600464
            },
            "confidence": 0.79381454,
            "action": 0,
            "action_probability": 1.0
        }
    }
}
```

</details>

### Architecture details

Laya uses a ModernBERT encoder followed by a task-specific decision head. Each question is converted into a prompt containing the input state, question instructions, and one `[MASK]` marker for every candidate option. The encoder produces contextual hidden states for the complete prompt.

The decision head adds a learned embedding for the question type, then applies a configurable stack of transformer layers. A scalar scorer produces one logit at each option marker. After temperature scaling and softmax, these logits become the option probabilities. The supported question types are:

- `choice`: select one option from a labelled set.
- `score`: predict a distribution over ordered levels and return its expected score.
- `noul`: predict whether a statement holds, represented as `false` or `true`.

The model also includes an action head. It combines the `[CLS]` representation with four distribution features: the maximum option probability, the gap between the top two probabilities, normalized entropy, and the number of options. The resulting action probabilities expose the model's recommended action and its probability alongside the decision.

### References

GitHub: https://github.com/NandhaKishorM/laya
HuggingFace: https://huggingface.co/convaiinnovations/laya
Architecture explanation: https://dev.to/nandakishor_m_6cc0adfde9f/i-built-non-autoregressive-decision-models-a-year-ago-then-a-frontier-lab-called-it-a-18me#:~:text=3.%20Model%20Architecture%3A%20421M%20Parameters
Paper 1: [SalesRLAgent: A reinforcement learning approach for real-time sales conversion prediction and optimization](https://arxiv.org/abs/2503.23303)
Paper 2: [Confidence-Aware Routing for Large Language Model Reliability Enhancement: A Multi-Signal Approach to Pre-Generation Hallucination Mitigation.](https://arxiv.org/abs/2510.01237)

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/text-embeddings-inference/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs).
- [x] Did you write any new necessary tests? If applicable, did you include or update the `insta` snapshots?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by vincevannoort** [comment]
Hey @alvarobartt,

I have a draft pull request to add support for structured decision models. Do you think there is any interest in getting this type of model included within TEI? The pull request will need some additional cleanup, possible split into smaller parts, etc. But I would like to check with you whether there is any interest for this getting included.

Thanks!
